internal/provider/integration_azure_resource.go

```diff
@@ -0,0 +1,267 @@
+			},
+			"scan_vms": schema.BoolAttribute{
+				MarkdownDescription: "Scan VMs.",
+				Optional:            true,
+			},
+			"subscription_allow_list": schema.ListAttribute{
+				MarkdownDescription: "List of Azure subscriptions to scan.",
+				Optional:            true,
+				ElementType:         types.StringType,
+			},
+			"subscription_deny_list": schema.ListAttribute{
+				MarkdownDescription: "List of Azure subscriptions to exclude from scanning.",
+				Optional:            true,
+				ElementType:         types.StringType,
+			},
+			"credentials": schema.SingleNestedAttribute{
+				Required: true,
+				Attributes: map[string]schema.Attribute{
+					"pem_file": schema.StringAttribute{
+						MarkdownDescription: "PEM file for Azure integration.",
+						Required:            true,
+						Sensitive:           true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *integrationAzureResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*mondoov1.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = &ExtendedGqlClient{client}
+}
+
+func (r *integrationAzureResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	var data integrationAzureResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Do GraphQL request to API to create the resource.
+	spaceMrn := ""
+	if data.SpaceId.ValueString() != "" {
+		spaceMrn = spacePrefix + data.SpaceId.ValueString()
+	}
+
+	var listAllow []mondoov1.String
+	allowlist, _ := data.SubscriptionAllowList.ToListValue(ctx)
+	allowlist.ElementsAs(ctx, &listAllow, true)
+
+	var listDeny []mondoov1.String
+	denylist, _ := data.SubscriptionDenyList.ToListValue(ctx)
+	denylist.ElementsAs(ctx, &listDeny, true)
+
+	// Check if both whitelist and blacklist are provided
+	if len(listDeny) > 0 && len(listAllow) > 0 {
+		resp.Diagnostics.AddError("ConflictingAttributesError", "Both subscription_allow_list and subscription_deny_list cannot be provided simultaneously.")
+		return
+	}
+
+	integration, err := r.client.CreateIntegration(ctx,
+		spaceMrn,
+		data.Name.ValueString(),
+		mondoov1.ClientIntegrationTypeAzure,
+		mondoov1.ClientIntegrationConfigurationInput{
+			AzureConfigurationOptions: &mondoov1.AzureConfigurationOptionsInput{
+				TenantID:               mondoov1.String(data.TenantId.ValueString()),
+				ClientID:               mondoov1.String(data.ClientId.ValueString()),
+				SubscriptionsWhitelist: &listAllow,
+				SubscriptionsBlacklist: &listDeny,
```

Comment:
# #@chris-rock I made the requested changes, but I don't think I can change the `SubscriptionsWhitelist` & `SubscriptionsBlackList` here since they are part of the GraphQL struct:
```
// AzureConfigurationOptionsInput
type AzureConfigurationOptionsInput struct {
	// (Required.)
	TenantID String `json:"tenantId"`
	// (Required.)
	ClientID String `json:"clientId"`

	// (Optional.)
	SubscriptionsWhitelist *[]String `json:"subscriptionsWhitelist,omitempty"`
	// (Optional.)
	SubscriptionsBlacklist *[]String `json:"subscriptionsBlacklist,omitempty"`
	// (Optional.)
	Certificate *String `json:"certificate,omitempty"`
	// (Optional.)
	ScanVms *Boolean `json:"scanVms,omitempty"`
	// (Optional.)
	ClientSecret *String `json:"clientSecret,omitempty"`
}
```

Reply:
Yes, that requires some API changes which we will do as well.